packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -1007,6 +1037,94 @@ contract OPContractsManagerUpgrader is OPContractsManagerBase {
         // Set the new implementation.
         setDGFImplementation(dgf, _gameType, IDisputeGame(newGame));
     }
+
+    /// @notice Sets the latest dispute game v2 implementations
+    /// @param _impls The container for the new dispute game implementations.
+    /// @param _l2ChainId The L2 chain ID
+    /// @param _disputeGame The current dispute game implementation in the dispute game factory
+    /// @param _newDelayedWeth The new delayed WETH implementation
+    /// @param _newAnchorStateRegistryProxy The new anchor state registry proxy
+    /// @param _gameType The type of game to deploy
+    /// @param _opChainConfig The OP chain configuration
+    function setNewGameImplV2(
+        OPContractsManager.Implementations memory _impls,
+        uint256 _l2ChainId,
+        IDisputeGame _disputeGame,
+        IDelayedWETH _newDelayedWeth,
+        IAnchorStateRegistry _newAnchorStateRegistryProxy,
+        GameType _gameType,
+        OPContractsManager.OpChainConfig memory _opChainConfig
+    )
+        internal
+    {
+        // If the prestate is set in the config, use it. If not set, we'll try to use the prestate
+        // that already exists on the current dispute game.
+        Claim absolutePrestate;
+        if (Claim.unwrap(_opChainConfig.absolutePrestate) == bytes32(0)) {
+            absolutePrestate = getAbsolutePrestate(
+                IDisputeGameFactory(_opChainConfig.systemConfigProxy.disputeGameFactory()),
+                address(_disputeGame),
+                _gameType
+            );
+        } else {
+            absolutePrestate = _opChainConfig.absolutePrestate;
+        }
+
+        // As a sanity check, if the prestate is zero here, revert.
+        if (absolutePrestate.raw() == bytes32(0)) {
+            revert OPContractsManager.PrestateNotSet();
+        }
+
+        IDisputeGame newGame;
+        bytes memory gameArgs;
+        if (GameType.unwrap(_gameType) == GameType.unwrap(GameTypes.PERMISSIONED_CANNON)) {
+            newGame = IDisputeGame(_impls.permissionedDisputeGameV2Impl);
+            address proposer = getProposer(IPermissionedDisputeGame(address(_disputeGame)));
+            address challenger = getChallenger(IPermissionedDisputeGame(address(_disputeGame)));
```

Comment:
These are just calling `_disputeGame.proposer()` and `.challenger()` which will return 0 for the implementation contract because they're now read from the game args.  For v2 contracts we need to decode the game args to get these values.
Although for the next upgrade we'd be upgrading from V1 contracts this would work, but then we probably have a footman for the next release so should use the same pattern as for `getAbsolutePrestate`.

Reply:
I've made the change now to support V2 contracts. Using a similar logic to `getAbsolutePrestate()`.
Do you think the toggle to use the gameArgs be behind a feature flag? Since the `gameArgs` will be empty for the next upgrade, the logic that reads from gameArgs will be unreachable until the upgrade after the next. So I'm OK with not gating that logic behind `DevFeatures`.

Maybe @smartcontracts has some thoughts on this.